File: notes.c
// c notes
#include <stdio.h>

void greet(char *name) {
    printf("hello %s\n", name);
}

int main() {
    greet("practice");

    int nums[3] = {1, 2, 3};
    for (int i = 0; i < 3; i++) {
        printf("%d\n", nums[i] * 2);
    }
    return 0;
}
// TODO: try pointers deeper next

File: notes.ts
// typescript notes
interface User {
    name: string;
    age: number;
}

const user: User = { name: "practice", age: 1 };

function greet(u: User): string {
    return `hello ${u.name}`;
}

console.log(greet(user));

// generics practice
function identity<T>(x: T): T {
    return x;
}
// TODO: try utility types next
